data_processing/wdl/workflows/hifi_qc_workflow.wdl

```diff
@@ -2,50 +2,137 @@ version 1.0
-            sample_id      = sample_id,
-            read_1_type    = "hifi",
-            read_2_type    = "other"
+            input_reads_1 = [hifi_reads],
+            input_reads_2 = other_reads,
+            sample_id = sample_id,
+            read_1_type = "hifi",
+            read_2_type = "other"
     }
 
+    # Optional methylation check
+    if (perform_methylation_check) {
+        call meth_check.methylationWorkflow as methylation {
+            input:
+                bam_files = [hifi_reads],
+                sample_id = sample_id
+        }
+    }
+    
+
+    call summarize_hifi_qc {
+        input:
+            ntsm_output = ntsm_wf.ntsm_eval_out,
+            readstat_report = read_stats.ReadStatsReport,
+            file_name = sample_id,
+            methylation_report = select_first([if (perform_methylation_check) then methylation.methylation_outputs else []])
+    }
 
     output {
         ## readstats output
-        File hifi_readstat_tar    = read_stats.ReadStatsTarball 
+        File hifi_readstat_tar = read_stats.ReadStatsTarball 
         File hifi_readstat_report = read_stats.ReadStatsReport
-
+        
         ## ntsm output
-        File hifi_ntsm_counts     = ntsm_wf.ntsm_count_1
-        File ext_ntsm_counts      = ntsm_wf.ntsm_count_2
-        File ntsm_eval            = ntsm_wf.ntsm_eval_out
+        File hifi_ntsm_counts = ntsm_wf.ntsm_count_1
+        File ext_ntsm_counts = ntsm_wf.ntsm_count_2
+        File ntsm_eval = ntsm_wf.ntsm_eval_out
+        File hifi_qc_summary = summarize_hifi_qc.summary_file
+        Array[File]? methylation_report = if (perform_methylation_check) then methylation.methylation_outputs else []
     }
 }
 
+task summarize_hifi_qc {
+    input {
+        File ntsm_output
+        File readstat_report
+        String file_name
+        Array[File]? methylation_report
+
+        Int memSizeGB = 4
+        Int threadCount = 1
+        Int disk_size = 16        
+        Int preempts = 2
+    }
+
+    command <<<
+		set -eux -o pipefail
+
+		# Define file paths
+		readstat_file="~{readstat_report}"
+		ntsm_file="~{ntsm_output}"
+		output_file="~{file_name}.summary.tsv"
+
+		# Extract values from the TSV file using awk
+		total_reads=$(awk -F'\t' 'NR==2 {print $3}' ${readstat_file})
+		total_bp=$(awk -F'\t' 'NR==3 {print $3}' ${readstat_file})
+		total_Gbp=$(awk -F'\t' 'NR==4 {print $3}' ${readstat_file})
+		total_min=$(awk -F'\t' 'NR==5 {print $3}' ${readstat_file})
+		total_max=$(awk -F'\t' 'NR==6 {print $3}' ${readstat_file})
+		total_mean=$(awk -F'\t' 'NR==7 {print $3}' ${readstat_file})
+		quartile_25=$(awk -F'\t' 'NR==8 {print $3}' ${readstat_file})
+		quartile_50=$(awk -F'\t' 'NR==9 {print $3}' ${readstat_file})
+		quartile_75=$(awk -F'\t' 'NR==10 {print $3}' ${readstat_file})
+		N25=$(awk -F'\t' 'NR==11 {print $3}' ${readstat_file})
+		N50=$(awk -F'\t' 'NR==12 {print $3}' ${readstat_file})
+		N75=$(awk -F'\t' 'NR==13 {print $3}' ${readstat_file})
```

Comment:
in the future, don't be afraid to use a heredoc to insert some python code to make this kind of thing easier.